File: PokemonGoRaidBot/src/api/mc.raid.description.ts
import { McTelegramUser } from "./mc.telegram.user";
import { McLocation } from "./mc.location";
import { McExternalSource } from "./mc.external.source";
import { McPublicationEntry } from "./mc.publication.entry";

export class McRaidDescription {
    public UniqueID: string;
    public PublicID: string;
    public User: McTelegramUser;
    public Location: McLocation;
    public Address: string;
    public Raid: string;
    public Gym: string;
    public Alignment: string;
    public RaidUnlockTime: string;
    public RaidEndTime: string;
    public UpdateCount: number;
    public Remarks: string;
    public TelegramMessageID: string;
    public Sources: McExternalSource[];
    public Publications: McPublicationEntry[];
    public Valor: number;
    public Mystic: number;
    public Instinct: number;
    public Unknown: number;
    public Maybe: number;
    public Total: number;
}

File: PokemonGoRaidBot/Backend/API/McRaidDescription.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using RaidBot.Backend.Bot.PokemonRaidBot.Entities;
using System;
using System.Collections.Generic;

namespace RaidBot.Backend.API
{
    [JsonObject(MemberSerialization.OptOut)]
    public class McRaidDescription
    {
        public Guid UniqueID { get; set; }

        public string PublicID { get; set; }

        public McTelegramUser User { get; set; }

        public McLocation Location { get; set; }

        public String Address { get; set; }

        public string Raid { get; set; }

        public string Gym { get; set; }

        [JsonConverter(typeof(StringEnumConverter))]
        public Team Alignment { get; set; }

        [JsonConverter(typeof(IsoDateTimeConverter))]
        public DateTime RaidUnlockTime { get; set; }

        [JsonConverter(typeof(IsoDateTimeConverter))]
        public DateTime RaidEndTime { get; set; }

        public int UpdateCount { get; set; }

        public string Remarks { get; set; }

        public long? TelegramMessageID { get; set; }

        public int Valor { get; set; }

        public int Mystic { get; set; }

        public int Instinct { get; set; }

        public int Unknown { get; set; }

        public int Maybe { get; set; }

        public List<McRaidPublication> Publications { get; set; }
    }

    public class McRaidPublication
    {
        public long ChannelID { get; set; }

        public long TelegramMessageID { get; set; }

        [JsonConverter(typeof(IsoDateTimeConverter))]
        public DateTime LastModificationTimeUTC { get; set; }
    }
}
